api/renewer_test.go

```diff
@@ -246,18 +245,18 @@ func TestLifetimeWatcher(t *testing.T) {
 // and therefore should be tested rigorously.
 func TestCalcSleepPeriod(t *testing.T) {
 	c := quick.Config{
-		MaxCount: 1000,
+		MaxCount: 10000,
 		Values: func(values []reflect.Value, r *rand.Rand) {
-			leaseDuration := r.Intn(math.MaxInt64)
-			remainingLeaseDuration := r.Intn(leaseDuration)
+			leaseDuration := r.Int63()
+			remainingLeaseDuration := r.Int63n(leaseDuration)
 			priorDuration := remainingLeaseDuration
-			increment := r.Intn(leaseDuration + 1)
+			increment := r.Intn(int(leaseDuration))
```

Comment:
I don't think you can safely cast leaseDuration to int, at least not a 32bit arch.  Why not use Int63n here too?